temoa/components/reserves.py

```
@@ -29,348 +44,447 @@
-        for (t, v) in model.process_reserve_periods[r, p]
-        if t in model.tech_storage
-        for i in model.process_inputs[r, p, t, v]
-        for o in model.process_outputs_by_input[r, p, t, v, i]
-    )
 
-    # The above code does not consider exchange techs, e.g. electricity
-    # transmission between two distinct regions.
-    # We take exchange takes into account below.
-    # Note that a single exchange tech linking regions Ri and Rj is twice
-    # defined: once for region "Ri-Rj" and once for region "Rj-Ri".
+def initialize_reserve_margins(model: TemoaModel) -> None:
```

Ruff (0.16.1)

[warning] Too many branches (16 > 12)

(PLR0912)